userspace/falco/app/restart_handler.cpp

```diff
@@ -31,8 +31,9 @@ limitations under the License.
 
 falco::app::restart_handler::~restart_handler()
 {
-    close(m_inotify_fd);
     stop();
+    close(m_inotify_fd);
```

Comment:
Close inotify_fd after having stopped the secondary thread (so that nobody is using it).
Also, reset it to -1.
/cc @jasondellaluce